src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaApiSecurityConfig.java

```diff
@@ -92,8 +107,43 @@ private AuthenticationProvider anonymousAuthenticationProvider() {
 		return anonymousAuthenticationProvider;
 	}
 
+
+	/**
+	 * Builds and returns an {@link AuthenticationProvider} based on the irida.administrative.authentication.mode config option
+	 *
+	 * @return {@link AuthenticationProvider}
+	 */
 	@Bean
 	public AuthenticationProvider authenticationProvider() {
+		IridaLdapSecurityConfig iridaLdapSecurityConfig = new IridaLdapSecurityConfig();
```

Comment:
Calling `new IridaLdapSecurityConfig()` means that Spring won't be managing this class. Instead you could try to `@Import({ IridaLdapSecurityConfig.class })` at the top of this config file.  That way it'll be managed by Spring.  You'll need to do some refactors on how this `public AuthenticationProvider authenticationProvider` method works, but that should help with the `NullPointerException` issue you were having on startup.

Reply:
Fixed as per conversation